miio/walkingpad.py

```diff
@@ -0,0 +1,189 @@
+import logging
+from collections import defaultdict
+from typing import Any, Dict
+
+import click
+
+from .click_common import command, format_output
+from .device import Device, DeviceStatus
+from .exceptions import DeviceException
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class WalkingpadException(DeviceException):
+    pass
+
+
+class WalkingpadStatus(DeviceStatus):
+    """Container for status reports from Xiaomi Walkingpad."""
+
+    # raw_command set_start_speed '["1.0"]' finally made my WalkingPad autostart on speed 1! Hurray!
+    #
+    # raw_command get_prop '["mode"]' returns
+    #
+    # [0] while mode is auto
+    #     [1] while mode is manual
+    #     [2] when in standby
+    # raw_command get_prop '["step"]' returns [2303]
+    # raw_command get_prop '["time"]' returns [1970] (while time is 32:50)
+    # raw_command get_prop '["dist"]' returns [1869]
+    # raw_command get_prop '["cal"]' returns [67340]
+    # raw_command get_prop '["goal"]' returns [0, 60]
+    # raw_command get_prop '["max"]' returns [6.0]
+    # raw_command get_prop '["initial"]' returns [3]
+    # raw_command get_prop '["offline"]' returns [0]
+    # raw_command get_prop '["sensitivity"]' returns [2]
+    # raw_command get_prop '["sp"]' returns [1.0]
+    # raw_command get_prop '["start_speed"]' returns [1.0]
+    # raw_command get_prop '["auto"]' returns [1]
+    # raw_command get_prop '["disp"]' returns [19]
+    # raw_command get_prop '["lock"]' returns [0]
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+
+        # NOTE: Only 1 property can be requested at the same time
+        self.data = data
+
+    @property
+    def power(self) -> str:
+        """Power state."""
+        return self.data["power"]
+
+    @property
+    def is_on(self) -> bool:
+        """True if the device is turned on."""
+        return self.power == "on"
+
+    @property
+    def time(self) -> int:
+        """Current lock status."""
+        return self.data["time"]
+
+    @property
+    def speed(self) -> float:
+        """Current speed."""
+        return self.data["sp"]
+
+    @property
+    def mode(self) -> int:
+        """Current mode."""
+        return self.data["mode"]
+
+    @property
+    def step(self) -> int:
```

Comment:
```suggestion
    def step_count(self) -> int:
```
Instead of using directly the names provided in the payloads, it makes sense to rename them to be comprehensible. Here one example :-)

Reply:
Done